pkg/exchange/binance/stream.go

```diff
@@ -207,6 +216,248 @@ func NewStream(client *binance.Client) *Stream {
 		}
 	})
 
+	stream.OnContinuousKLineEvent(func(e *ContinuousKLineEvent) {
+		kline := e.KLine.KLine()
+		if e.KLine.Closed {
+			stream.EmitContinuousKLineClosedEvent(e)
+			stream.EmitKLineClosed(kline)
+		} else {
+			stream.EmitKLine(kline)
+		}
+	})
+
+	stream.OnDisconnect(func() {
+		log.Infof("resetting depth snapshots...")
+		for _, f := range stream.depthFrames {
+			f.emitReset()
+		}
+	})
+
+	stream.OnConnect(func() {
+		var params []string
+		for _, subscription := range stream.Subscriptions {
+			params = append(params, convertSubscription(subscription))
+		}
+
+		if len(params) == 0 {
+			return
+		}
+
+		log.Infof("subscribing channels: %+v", params)
+		err := stream.Conn.WriteJSON(StreamRequest{
+			Method: "SUBSCRIBE",
+			Params: params,
+			ID:     1,
+		})
+
+		if err != nil {
+			log.WithError(err).Error("subscribe error")
+		}
+	})
+
+	return stream
+}
+
+func NewFuturesStream(client *futures.Client) *Stream {
```

Comment:
a different constructor but returns the same Stream struct?

Reply:
Yes, it is.

Because the code where you mentioned above return global type stream at exchange.go line 320.
These two different stream will return client and futuresClient, respectively, in the same global stream.